src/hub/ShareClassManager.sol

```diff
@@ -312,6 +323,58 @@ contract ShareClassManager is Auth, IShareClassManager {
         );
     }
 
+    /// @inheritdoc IShareClassManager
+    function updatePricePerShare(PoolId poolId, ShareClassId scId_, D18 navPoolPerShare) external auth {
+        require(exists(poolId, scId_), ShareClassNotFound());
+
+        ShareClassMetrics storage m = metrics[scId_];
+        m.navPerShare = navPoolPerShare;
+        emit UpdateShareClass(
+            poolId, scId_, navPoolPerShare.mulUint128(m.totalIssuance), navPoolPerShare, m.totalIssuance
```

Comment:
NIT

Can you remove the `navPoolPerShare.mulUint128(m.totalIssuance)` and the `m.totalIssuance`? Those will not be ensured to be up-to-date. I think that is misleading. But not required.

Reply:
Fixed